### DISCUSSION TITLE
unbinding let

### DISCUSSION THREAD

**Discussion by allison-casey** [hylang/hy#2193] [Show and tell]
`let` is now natively in Hy! 

Native `let` (like the `let` of old) doesn't garbage collect let bound names after you exit scope, but the macro to do so ended up just kinda working as you'd expect now that let just assigns to gensym names instead of a dictionary. 

```clojure
(defmacro let-del [bindings #* body]
  (let [result (hy.gensym)]
    `(let ~bindings
       (setv ~result (do ~@body))
       ~@(gfor binding (cut bindings None None 2) `(del ~binding))
       ~result)))
``` 
All hail native `let`! 

note though that deleting bindings like this will break closures in the same way
deleting any vars used in a closure would in python :woman_shrugging: 

Make sure to thank @scauligi for all their hard work getting `let` into Hy :heart:

**Comment by dpavlic** [comment]
This is a very stupid question, but if I am using let in a defmacro, will the let bindings all be out of scope and be eligible for GC after the macro call? Or do I still have to manually del them or use the macro above?
